### PR TITLE
Update synthetic access

### PR DESCRIPTION
**Pull request by doncem**
Notes:

- include public access to headers
- better documentation of the synthetic initialiser

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,13 +56,13 @@ Maven:
...
 </dependency>
 ```
 
 Gradle:
 ```groovy
-compile group: 'uk.gov.hmcts.reform', name: 'java-logging-httpcomponents', version: '2.2.0'
+compile group: 'uk.gov.hmcts.reform', name: 'java-logging-httpcomponents', version: '2.2.1'
 ```
 
 **Please note:** You will also need to implement a class that configures an HTTP client with interceptors for outbound HTTP requests and responses. See https://github.com/hmcts/cmc-claim-store/blob/master/src/main/java/uk/gov/hmcts/cmc/claimstore/clients/RestClient.java#L98 for an example.
```
Comment (nitinprabhu) [inline README.md]: Not part of this PR but `https://github.com/hmcts/cmc-claim-store/blob/master/src/main/java/uk/gov/hmcts/cmc/claimstore/clients/RestClient.java#L98` doesn't work

File: java-logging-appinsights/README.md [hunk 2]
```diff
@@ -97,21 +97,18 @@ For custom telemetry metrics implement `AbstractAppInsights` already provided wi
 - CloudRoleNameInitializer
 
 `WebSyntheticRequestTelemetryInitializer` provides functionality to separate out requests via so called Synthetic Source tag.
```
Comment (nitinprabhu) [inline java-logging-appinsights/README.md]: where do we add `SyntheticTest-Source` in the config?

File: java-logging-appinsights/README.md [hunk 3]
```diff
@@ -97,21 +97,18 @@ For custom telemetry metrics implement `AbstractAppInsights` already provided wi
...
-  - `Application Insights Availability Monitoring` as Source name
-  - `SyntheticTest-RunId`
-  - `SyntheticTest-Location`
+All headers are available in helper class [SyntheticHeaders](java-logging-appinsights/src/main/java/uk/gov/hmcts/reform/logging/appinsights/SyntheticHeaders).
+All headers are optional and by default will be assumed `Application Insights Availability Monitoring` as Synthetic source name.
+Additionally default setup tries to include `SyntheticTest-RunId` and `SyntheticTest-Location` if present.
+For particular requests required to be separated, case can provide additional header `SyntheticTest-Source` which will replace default source name.
+Along with custom Source name there are more optional headers to be applied at will:
+
+- `SyntheticTest-UserId`
```
Comment (nitinprabhu) [inline java-logging-appinsights/README.md]: where do we add headers`SyntheticTest-Source` in the config?

File: README.md (hunk 1 above)
Comment (doncem) [reply README.md]: Oogh. This readme needs a better reviewer from the relevant team

File: java-logging-appinsights/README.md [hunk 4]
```diff
@@ -97,21 +97,18 @@ For custom telemetry metrics implement `AbstractAppInsights` already provided wi
...
-  - `SyntheticTest-SessionId`
-  - `SyntheticTest-OperationId`
-  - `SyntheticTest-TestName`
-  - `SyntheticTest-RunId`
-  - `SyntheticTest-Location`
-- Otherwise:
-  - `Application Insights Availability Monitoring` as Source name
-  - `SyntheticTest-RunId`
-  - `SyntheticTest-Location`
+All headers are available in helper class [SyntheticHeaders](java-logging-appinsights/src/main/java/uk/gov/hmcts/reform/logging/appinsights/SyntheticHeaders).
```
Comment (banderous) [inline java-logging-appinsights/README.md]: Would be nice to link to javadoc instead of having to repeat values inside this readme.

Comment (banderous) [reply java-logging-appinsights/README.md]: One day

File: README.md [hunk 5]
```diff
@@ -127,7 +127,7 @@ Which produces:
   "environment" : "undefined"
 }
 {
-  "timestamp" : "2017-02-02T12:32:38.252+00:00",
+  "timestamp" : "2017-02-02T12:32:38.2.2.10:00",
```
Comment (banderous) [inline README.md]: What's happening here?

Comment (doncem) [reply README.md]: Dear me! Bogus shell script for version update. Will include a fix in this PR as it's a super minor
